api/src/main/java/org/infinispan/api/reactive/Counter.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.api.reactive;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 13.0
+ **/
+public interface Counter {
+
+   CompletionStage<Long> value();
```

Comment:
if we do reactive, unless something has changed we need to use mutiny, otherwise api integration with other frameworks will be a big pain. maybe we may want to ping Clément for this as well?

Reply:
That's what I have already stated in the description. I have now pushed a skeleton mutiny-based reactive api